polaris-react/src/components/DropZone/DropZone.tsx

```diff
@@ -427,6 +430,7 @@ export const DropZone: React.FunctionComponent<DropZoneProps> & {
               type="file"
               ref={inputRef}
               autoComplete="off"
+              tabIndex={onClick && dragAndDropOnly ? -1 : 0}
```

Comment:
We need to prevent the input from being visible to screenreaders if it cannot be reached.

```suggestion
              aria-hidden={dragAndDropOnly ? true : undefined}
              tabIndex={onClick && dragAndDropOnly ? -1 : 0}
```